airflow/www/views.py

```diff
@@ -1725,14 +1725,25 @@ def index(self):
         qry = None
         # filter the dags if filter_by_owner and current user is not superuser
         do_filter = FILTER_BY_OWNER and (not current_user.is_superuser())
+        owner_mode = conf.get('webserver', 'OWNER_MODE').strip().lower()
+
         if do_filter:
-            qry = (
-                session.query(DM)
-                    .filter(
-                    ~DM.is_subdag, DM.is_active,
-                    DM.owners == current_user.username)
-                    .all()
-            )
+            if owner_mode == 'ldapgroup':
```

Comment:
This can be refactored to be a bit cleaner and not repeat code too much. qry = qry.filter()